### PR TITLE
Feature/add tap event

### LINKED ISSUE #287: Add cross interface tap event type
To improve mobile responsiveness to `click` events we should introduce a custom cross-interface `tap` event type.

- [x] Should not be ignored by `passive` events - except if specified
- [ ] should be able to `preventDefault` click actions
- [x] Should support plug and play of touch interfaces
- [x] needs to works with **Touch Events**, **Pointer Events** and classic **Mouse Events**
- [x] We should never need to use any `cursor: pointer` hacks to enable them
- [ ] Scroll needs to be prevented in case of a valid tap
- [x] mitigates click delay #20 
- [ ] **IMPORTANT** remove this iOS shame to make outer click events work:
   - delete [`outer-ios-SHAME.js`](https://github.com/axa-ch/patterns-library/blob/master/src/js/outer-ios-SHAME.js)
   - remove it's usage from [`outer.js`](https://github.com/axa-ch/patterns-library/blob/master/src/js/outer.js#L19)

### PR DESCRIPTION
**Pull request by AndyOGo**
fixes #287, #20

### CODE REVIEW

File: src/js/tap.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+const regexPointer = /touch|pointer/;
+const TAP_THRESHOLD = 150;
+const TAP_PRECISION = 60 / 2;
+const getPointer = event => event.targetTouches ? event.targetTouches[0] : event;
+
+// eslint-disable-next-line consistent-return
+function addEventListenerProxy(eventType, ...args) {
+  const eventTarget = this;
+
+  if (!hasSupport) {
```
Comment (LucaMele) [inline src/js/tap.js]: unnecessary lines if u add `eventType = 'click'` on line 23

File: src/js/tap.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    eventType = 'click'; // eslint-disable-line no-param-reassign
+  }
+
+  if (eventType !== 'tap') {
+    return _addEventListener.call(eventTarget, eventType, ...args);
+  }
+
+  console.log(`>>> proxy ${eventType}`);
```
Comment (LucaMele) [inline src/js/tap.js]: ?

File: src/js/tap.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+    if (isValidThreshold && isValidPrecision) {
+      fire(target, eventType, event);
+    }
+  }
+}
+
+// eslint-disable-next-line consistent-return
+function removeEventListenerProxy(eventType, ...args) {
+  if (!hasSupport) {
```
Comment (LucaMele) [inline src/js/tap.js]: dtto

**Comment by AndyOGo** [comment]
@LucaMele 
The custom `tap` event type works now, but due to our multiple loading of modules. It is bound X times.
This needs to be fixed by properly bundling/loading modules.

tap-bound-multiple-times

**Comment by LucaMele** [comment]
bildschirmfoto 2018-06-06 um 09 38 38

**Comment by AndyOGo** [comment]
@LucaMele 
It's apparently an infinite loop, though the real problem is that this module is define multiple times (once for each web component). Which causes it to be patched multiple times und thus it logs very often for a single call.
It's the same problem we have with `on`.
Will be fixed until we have proper builds including code-splitting.

**Comment by LucaMele** [comment]
it makes browser to freeze thought.

**Comment by AndyOGo** [comment]
yes I now and max call stack...

**Comment by AndyOGo** [comment]
**Breaking Changes:** Please migrate before merging
https://github.com/axa-ch/patterns-library/releases/tag/v2.0.1-beta.217

**Pull request closed without merging**